src/main/java/com/monumental/services/AsyncMonumentService.java

```diff
@@ -0,0 +1,100 @@
+package com.monumental.services;
+
+import com.amazonaws.SdkClientException;
+import com.monumental.models.Image;
+import com.monumental.models.Monument;
+import com.monumental.repositories.MonumentRepository;
+import com.monumental.util.async.AsyncJob;
+import com.monumental.util.csvparsing.CsvMonumentConverterResult;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Service;
+
+import java.io.File;
+import java.util.*;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * This class is the async counterpart to the MonumentService, where any asynchronous tasks
+ * should be kept.
+ */
+@Service
+public class AsyncMonumentService extends AsyncService {
+
+    @Autowired
+    MonumentRepository monumentRepository;
+
+    @Autowired
+    TagService tagService;
+
+    @Autowired
+    AwsS3Service s3Service;
+
+    /**
+     * Insert validated monument CSV rows and related objects
+     * @param job - The AsyncJob to report progress to
+     * @param csvResults - The validated CSV rows, converted into monuments
+     * @return List of inserted monuments
+     */
+    @Async
+    public CompletableFuture<List<Monument>> bulkCreateMonuments(AsyncJob job, List<CsvMonumentConverterResult> csvResults) {
```

Comment:
Since this is marked as `@Async`, Spring will do a bunch of proxy magic to make this run in a separate thread. No need for us to mess with that stuff 🙌